samples/bootloader/README.rst

```diff
@@ -0,0 +1,170 @@
+.. _bootloader:
+
+Immutable bootloader
+####################
+
+The bootloader sample implements an immutable first stage bootloader that has the capability to verify and boot a second stage bootloader.
+If the second stage bootloader is upgradable, this sample also provides a mechanism to upgrade the second stage bootloader.
+
+See :ref:`ug_bootloader` for more information about the full bootloader chain.
+
+Overview
+********
+
+The bootloader sample provides a simple root of trust (RoT) anchor as well as support for an upgradable second stage bootloader.
+
+This is accomplished by the following steps:
+
+1. Lock the flash.
+     To enable the RoT, the bootloader sample locks the flash that contains the sample bootloader and its configuration.
+     Locking is done using the hardware that is available on the given architecture.
+     For details on locking, see the :ref:`fprotect_readme` driver.
+
+#. Select the next stage in the boot chain.
+     When the bootloader sample is enabled, there are two slots in which the second stage bootloader can reside.
+     The second stage bootloader in each slot has a version number associated with it, and the bootloader sample selects the second stage bootloader that has the highest version number.
+
+#. Verify the next stage in the boot chain.
+     After selecting the image to be booted next, the bootloader sample verifies its validity using one of the provisioned public keys hashes.
+     The next boot stage has a set of metadata associated with it.
+     This metadata contains the full public key corresponding to the private key that was used to sign the firmware.
+     To save space, the bootloader sample checks only the hash of this public key against a set of provisioned hashes.
+     If the public key in the metadata matches one of the provisioned public key hashes, the image is considered valid.
+     If the public key does not match any of the provisioned hashes, validation fails.
+
+#. Boot the next stage in the boot chain.
+    After verifying the next boot stage, the bootloader sample uninitializes all peripherals that it used and boots the next boot stage.
+
+#. Share the cryptographic library over ABI.
+     The bootloader sample does not contain any flash writing code.
+     Therefore, it is safe to share some of its functionality through an Application Binary Interface (ABI, see :ref:`fw_info_abi`).
```

Comment:
This link will cause the doc build to break until we merge https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1291